katas/content/qec_shor/index.md

```diff
@@ -0,0 +1,371 @@
+})
+
+This kata introduces you to the basic concepts of quantum error correction using several simple error correction codes.
+
+**This kata covers the following topics:**
+
+- simple models of noise in quantum systems,
+- parity measurements in Z and X bases,
+- bit flip code - the simplest code that protects qubits from the effects of bit flip noise,
+- phase flip code - the simplest code that protects qubits from the effects of phase flip noise,
+- Shor code - the simplest code that can protect from an arbitrary error on a single qubit.
+
+**What you should know to start working on this kata:**
+
+- Basic single-qubit and multi-qubit gates
+- Single-qubit and multi-qubit quantum measurements and their effect on quantum systems
+
+@[section]({
+    "id": "qec_shor__noise",
+    "title": "Noise in Classical and Quantum Systems"
+})
+
+Any quantum systems we can use to carry out quantum computation are inherently noisy. 
+Quantum noise can be caused by different physical processes, depending on the type of a particle or device used as a qubit.
+From the computation point of view, the presence of noise in the quantum system means that its state can suffer from random errors, and thus end up different from the state we're relying on to do our computations. 
+This makes computations unreliable very fast, since the effects of noise accumulate quickly to make computation results effectively random.
+
+Before we dive into dealing with noise in quantum systems, let's take a quick look at how we do that for classical systems.
+
+The model used for analyzing classical noise is called *binary symmetric channel*, in which classical bits sent through the channel are transmitted correctly with probability $1-p$ and flipped with probability $p$.
+In this scenario, the information sent through the channel can be protected against the effects of the noise using *repetition code*:
+
+- On the sender side, we replace each bit we want to send with three copies of itself:
+$$0 \rightarrow 000, 1 \rightarrow 111$$
+- On the receiver side, we decode the original bit by majority vote: 
+$$000, 100, 010, 001 \rightarrow 0$$
+$$111, 011, 101, 110 \rightarrow 1$$
+
+What is the probability of this scheme failure, that is, the value of the message bit changing after it was sent through the channel? 
+Majority vote allows for one error on any bit to happen without affecting the decoding outcome, so it would take two or three errors happening on individual bits for decoding to produce an incorrect result. The probability of this happening is $3p^2(1-p) + p^3 = 3p^2 - 2p^3$. If we compare this with the probability of an individual bit transmission failing $p$, we can see that using repetition code yields higher success probability, as long as $p < \frac12$. We can improve success probability further by increasing the number of repetitions we use to encode each bit: $5$ repetitions allow us to detect and correct $2$ errors, $7$ repetitions - $3$ errors, and so on.
```

Comment:
```suggestion
Majority vote allows for one error on any bit to happen without affecting the decoding outcome, so it would take two or three errors happening on individual bits for decoding to produce an incorrect result. The probability of this happening is $3p^2(1-p) + p^3 = 3p^2 - 2p^3$. If we compare this with the probability of an individual bit transmission failing $p$, we can see that using the repetition code yields higher success probability, as long as $p < \frac12$. We can improve success probability further by increasing the number of repetitions we use to encode each bit: $5$ repetitions allow us to detect and correct $2$ errors, $7$ repetitions - $3$ errors, and so on.
```